django-nonrel/ocl/oclapi/models.py

```diff
@@ -74,6 +78,40 @@ def get_default_view_name(self):
         }
         return self._default_view_name % format_kwargs
 
+    def encode_extras(self):
+        logger.debug('extras:' + str(self.extras))
+        if self.extras is not None and self.extras_are_encoded is False:
+            encoded_extras = {}
+            extras = self.extras
+            for old_key in extras:
+                logger.debug(old_key)
+                key = old_key
+                key = key.replace('%', '%25')
+                key = key.replace('.','%2E')
+                value = extras.get(old_key)
+                logger.debug(key)
+                encoded_extras[key] = value
+            self.extras = encoded_extras
+            self.extras_are_encoded = True
+        return
+
+    def decode_extras(self):
+        logger.debug("decoding extras")
+        if self.extras is not None and self.extras_are_encoded is True:
+            decoded_extras = {}
+            extras = self.extras
+            for old_key in extras:
+                logger.debug(old_key)
+                key = old_key
+                key = key.replace('%25', '%')
+                key = key.replace('%2E', '.')
+                value = extras.get(old_key)
+                logger.debug(key)
+                decoded_extras[key] = value
+            self.extras = decoded_extras
+            self.extras_are_encoded = False
+            logger.debug("extras decoded")
+        return
```

Comment:
Could you please write some tests for that in djanto-nonrel/ocl/concepts/tests.py?